subprojects/jacoco/src/integTest/groovy/org/gradle/testing/jacoco/plugins/JacocoAggregationIntegrationTest.groovy

```diff
@@ -0,0 +1,486 @@
+/*
+ * Copyright 2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.testing.jacoco.plugins
+
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+import org.gradle.testing.jacoco.plugins.fixtures.JacocoReportXmlFixture
+
+class JacocoAggregationIntegrationTest extends AbstractIntegrationSpec {
+    def setup() {
+        multiProjectBuild("root", ["application", "direct", "transitive"]) {
+            buildFile << """
+                allprojects {
```

Comment:
I know it's just to save time setting up tests, but don't we want to discourage use of the `allprojects` and `subprojects` blocks where possible?  It's a setup method, so one string for each project's buildscript might be clearer anyway?